azurelinuxagent/ga/exthandlers.py

```diff
@@ -1274,33 +1259,8 @@ def download(self):
                 self.logger.info("The existing extension package is invalid, will ignore it.")
 
         if not package_exists:
-            downloaded = False
```

Comment:
Download extension used to have its own retry logic. No longer needed. All the download logic is now in 1 single place and is consistent across artifact blob, manifests, and extension packages